src/platform/windows/gui/options/options.cpp

```diff
@@ -192,18 +193,42 @@ void drawImage(const std::optional<Image> image, float w = 0.f, float h = 0.f) {
     }
 }
 
-void button(int controller, const std::string& button, const char* tooltip = nullptr) {
+void pressKeyPopup() {
+    auto inputManager = static_cast<SdlInputManager*>(InputManager::getInstance());
+    if (ImGui::BeginPopupModal("Waiting for key...", &inputManager->waitingForKeyPress, ImGuiWindowFlags_AlwaysAutoResize)) {
+        ImGui::Text("Press any key (ESC to cancel).");
+        ImGui::EndPopup();
+    }
+}
+
+void button(const std::string& button, const char* tooltip, int controller = 0) {
     auto inputManager = static_cast<SdlInputManager*>(InputManager::getInstance());
     static std::string currentButton = "";
-    if (controller < 1 || controller > 4) return;
+    if (controller < 0 || controller > 4) return;
+    bool hotkey = controller == 0;
     int ctrl = controller - 1;
-
+    auto& relatedConfig = hotkey ? config.hotkeys[button] : config.controller[ctrl].keys[button];
+    
     if (button == currentButton && inputManager->lastPressedKey.type != Key::Type::None) {
-        config.controller[ctrl].keys[button] = inputManager->lastPressedKey.to_string();
+        relatedConfig = inputManager->lastPressedKey.to_string();
         inputManager->lastPressedKey = Key();
     }
 
-    std::string key = config.controller[ctrl].keys[button];
+    std::string key = relatedConfig;
+
+    key = Key(key).getName();
+
+    if (hotkey) {
+        if (ImGui::Button(fmt::format("{}##{}", key, button).c_str(), ImVec2(150.f, 0.f))) {
+            currentButton = button;
+            inputManager->waitingForKeyPress = true;
+            ImGui::OpenPopup("Waiting for key...");
+        }
+        ImGui::SameLine();
+        ImGui::Text(tooltip);
```

Comment:
```c++
/home/build/android/app/jni/../../../build/libs/avocado/../../../src/platform/windows/gui/options/options.cpp:228:21: error: format string is not a string literal (potentially insecure) [-Werror,-Wformat-security]
          ImGui::Text(tooltip);
```
Use 
```c++
ImGui::TextUnformatted
```

Reply:
It looks like a small error (ImGui::Text (tooltip) -> ImGui::Text ("%s",tooltip) ?). But unfortunately now I can only build and test on windows.

Reply:
Both `ImGui::TextUnformatted(tooltip)` and `ImGui::Text ("%s",tooltip)` would work in this case.
Dunno why only Android NDK compiler has this check enabled.